lib/core/src/stringOpr.cpp

```diff
@@ -45,28 +45,27 @@ char *rmemcpy( void *dest, const void *src, int strLen, int maxLen ) {
     }
 }
 
-char *rstrcpy( char *dest, const char *src, int maxLen ) {
+char* rstrcpy(char* dest, const char* src, int maxLen)
+{
     // snprintf with logging on fail
-
-    if ( dest == NULL || src == NULL ) {
-        return NULL;
+    if (dest == nullptr || src == nullptr) {
+        return nullptr;
     }
-    int status = snprintf( dest, maxLen, "%s", src );
 
-    if ( status >= 0 && status < maxLen ) {
-        return dest;
-    }
-    else if ( status >= 0 ) {
-        rodsLog( LOG_ERROR,
-                 "rstrcpy not enough space in dest, slen:%d, maxLen:%d, stacktrace:\n%s",
-                 status, maxLen, irods::stacktrace().dump().c_str());
-        rodsLog( LOG_DEBUG, "rstrcpy arguments dest [%s] src [%s]", dest, src );
-        return NULL;
-    }
-    else {
-        rodsLog( LOG_ERROR, "rstrcpy encountered an encoding error." );
-        return NULL;
+    int status{std::snprintf(dest, maxLen, "%s", src)};
+
+    if (status >= 0) {
+        if (status < maxLen) {
+            return dest;
+        }
+        rodsLog(LOG_ERROR,
+            "rstrcpy not enough space in dest, slen:%d, maxLen:%d; arguments dest [%s] src [%s]",
+            status, maxLen, dest, src);
```

Comment:
If possible, let's switch to the new logger.
```cpp
using log_server = irods::experimental::log::server;
...
log_server::error(<your-args-here>);
```

Also, consider replacing `dest`, `src`, `sLen` and `maxLen`. Those words are terrible for these types of log messages. We can be as descriptive as we want here.

Think on it.